tests/python_package_test/test_dask.py

```diff
@@ -262,6 +262,37 @@ def _unpickle(filepath, serializer):
         raise ValueError(f'Unrecognized serializer type: {serializer}')
 
 
+def _objective_least_squares(y_true, y_pred):
+    grad = y_pred - y_true
+    hess = np.ones(len(y_true))
+    return grad, hess
+
+
+def _objective_logistic_regression(y_true, y_pred):
+    y_pred = 1.0 / (1.0 + np.exp(-y_pred))
+    grad = y_pred - y_true
+    hess = y_pred * (1.0 - y_pred)
+    return grad, hess
+
+
+def _objective_logloss(y_true, y_pred):
+    num_rows = len(y_true)
+    num_class = len(np.unique(y_true))
+    # operate on preds as [num_data, num_classes] matrix
+    y_pred = y_pred.T.reshape(-1, num_class)
+    row_wise_max = np.max(y_pred, axis=1).reshape(num_rows, 1)
+    preds = y_pred - row_wise_max
+    prob = np.exp(preds) / np.sum(np.exp(preds), axis=1).reshape(num_rows, 1)
+    grad_update = np.zeros_like(preds)
+    grad_update[np.arange(num_rows), y_true.astype('int')] = -1.0
+    grad = prob + grad_update
```

Comment:
I believe this should be something like:
```suggestion
    ohe_classes = np.zeros_like(prob)
    ohe_classes[np.arange(num_rows), y_true.astype(int)] = 1.0
    grad = ohe_classes * (prob - 1) + (1 - ohe_classes) * prob
```

Reply:
hmmm are you sure? It seems to me that the existing current logic is correct, based on

https://github.com/microsoft/LightGBM/blob/af5b40e1f653e25a5f3470e09de4d1c68dc59663/src/objective/multiclass_objective.hpp#L100-L103

Looking more closely, it also seems like your proposed change and the existing code in this PR are identical. Aren't both saying "the gradient is `prob` for classes not equal to the label and otherwise `prob - 1`"?

Reply:
You're right, that's my bad haha.

Reply:
ha no problem!